Microsoft.Toolkit/Collections/ObservableGroup.cs

```diff
@@ -49,9 +52,21 @@ public ObservableGroup(TKey key, IEnumerable<TValue> collection)
         }
 
         /// <summary>
-        /// Gets the key of the group.
+        /// Gets or sets the key of the group.
         /// </summary>
-        public TKey Key { get; }
+        public TKey Key
+        {
+            get => this.key;
+            set
+            {
+                if (!EqualityComparer<TKey>.Default.Equals(this.key, value))
+                {
+                    this.key = value;
+                    
```

Comment:
The CI build is failing due to trailing whitespaces here 👍
```suggestion

```

Reply:
Done